src/esbmc/bmc.cpp

```diff
@@ -726,79 +730,104 @@ smt_convt::resultt bmct::multi_property_check(
-#endif
-#endif
-      // Set up the current claim and slice it!
-      claim_slicer claim(i);
-      claim.run(local_eq->SSA_steps);
-      symex_slicet slicer(options);
-      slicer.run(local_eq->SSA_steps);
-
-      // Initialize a solver
-      auto runtime_solver =
-        std::shared_ptr<smt_convt>(create_solver("", ns, options));
-      // Save current instance
-      generate_smt_from_equation(runtime_solver, local_eq);
-
-      log_status(
-        "Solving claim '{}' with solver {}",
-        claim.claim_msg,
-        runtime_solver->solver_text());
-
-      smt_convt::resultt result;
-      /* TODO: We might move this into solver_convt. It is
-       * useful to have the solver as a thread.
-       */
-      std::thread solver_job(
-        [&result, &runtime_solver]() { result = runtime_solver->dec_solve(); });
-
-      const bool fail_fast = options.get_bool_option("multi-fail-fast");
-      // This loop is mainly for fail-fast.
-      try
+    log_status(
+      "Solving claim '{}' with solver {}",
+      claim.claim_msg,
+      runtime_solver->solver_text());
+    total_instance++;
+
+    smt_convt::resultt result = runtime_solver->dec_solve();
+
+    const bool fail_fast = options.get_bool_option("multi-fail-fast");
+    // This try-catch is mainly for fail-fast.
+    //?TODO: "multi-fail-fast n": stop after first n SATs found.
+    try
+    {
+      if(result == smt_convt::P_SATISFIABLE)
       {
-        while(!solver_job.joinable())
+        if(fail_fast)
         {
-          // Try again 100ms later
-          std::this_thread::sleep_for(
-            std::chrono::duration<int, std::milli>(100));
-          // Did someone finished already?
-          if(fail_fast && final_result == smt_convt::P_SATISFIABLE)
+          throw 0;
+        }
+
+        bool is_compact_trace = true;
+        if(
+          options.get_bool_option("no-slice") &&
+          !options.get_bool_option("compact-trace"))
+          is_compact_trace = false;
+
+        goto_tracet goto_trace;
+        build_goto_trace(
+          local_eq, runtime_solver, goto_trace, is_compact_trace);
+
+        // Store the comment and location of the assertion
+        // to avoid double verifying the claims that are already verified
+        std::string cmt_loc = "";
+
+        for(const auto &step : goto_trace.steps)
+          if(step.type == goto_trace_stept::ASSERT)
           {
-            log_status("Other thread already found a SAT VCC.");
-            throw 0;
+            // since we only handle one claim at a time
+            // we will/should not overwrite the loc
+            assert(cmt_loc == "");
+            std::string loc;
+            if(step.pc->location.is_nil())
+              loc = "nil";
+            else
+              loc = step.pc->location.as_string();
+
+            // we use the "comment + location" to distinguish each claim
+            // - locaiton: ideally, the loc is enough for distinguishuing claims.
```

Comment:
Typos: location and distinguishing. Also: Why "ideally"? Is it or it is not enough to distinguish claims? I would actually be surprised if it was because we butcher the location somewhat in <https://github.com/esbmc/esbmc/blob/cc230ed73c5f7b51ec6381295ce07e151649fbc2/src/clang-c-frontend/clang_c_convert.cpp#L3421>.

Reply:
My assumption is that the location of each claim is unique ideally, as they have different line numbers. But some mutations/checkings in ESBMC (e.g. add-false-assert) will add the instructions with the same location as the previous instruction. 

Still, I think the comment is not clear enough and not necessary. The core idea should be:
- comment + location : `Claim x: ... location line y`
- x is unique for sure. However, the unwinding asserts do not have these `Claim x` prefixes. Therefore we add the location behind, as the line number for each unwinding assert is different.

Will Fix typo.

Reply:
Right, that makes more sense, thanks!